### DISCUSSION TITLE
Ability to mutate state in block component

### DISCUSSION THREAD

**Discussion by lotarbo** [filamentphp/filament#6076] [Help]
Hello. i have situation:
Page resource with many fields, and with field  builder `content` - which contains many blocks
In this blocks, i have `field media - Filament Curator.` this is a reusable field in the whole project.
But later, i change this field to `builder`, because i need). So in edit page i have error in this line https://github.com/filamentphp/filament/blob/2.x/packages/forms/src/Components/Builder.php#L290 it needed array and int given.
I try to use different Dehydration methods on `bulder` field, but it didn't work in this situation.

So i write a patch which change data in the whole database, but i think it's not the best idea.
I think better - add some mutation to` builder field `and transform data for correct format, but i dont know how to do it right now(

**Comment by danharrin** [comment]
Changing the data in the database to be the correct format is definitely the best idea, rather than having to do this every time the form is loaded and saved

**Reply by lotarbo** [reply]
this will only happen once, the first time the form is edited.
To change in database, i need to write complex script (

**Reply by danharrin** [reply]
Well that is my suggestion. Otherwise you are just asking for problems and inconsistency in the future.

**Reply by lotarbo** [reply]
ok, thank you) as i understand, there are no ability to do what i mean?

**Reply by danharrin** [reply]
You could use an Eloquent cast or something, but by this point you could have written the script 10 times 😅
